src/tests/end2end/D3D12ResidencyManagerTests.cpp

```diff
@@ -75,6 +75,106 @@ class D3D12ResidencyManagerTests : public D3D12TestBase, public ::testing::Test
         return (segment.Budget > segment.CurrentUsage) ? (segment.Budget - segment.CurrentUsage)
                                                        : 0;
     }
+
+    class CreateHeapNotResidentFnContext {
```

Comment:
Could we please re-use the same (internal) implementation here (and elsewhere) and just set the flags it needs?

Reply:
Done.